app/code/Magento/Cms/Model/Page/Service/PageService.php

```diff
@@ -0,0 +1,67 @@
+<?php
+declare(strict_types=1);
+/**
+ * Copyright 2015 Adobe
+ * All Rights Reserved.
+ */
+namespace Magento\Cms\Model\Page\Service;
+
+use Magento\Cms\Api\Data\PageInterface;
+use Magento\Cms\Api\PageRepositoryInterface;
+use Magento\Cms\Model\Page\CustomLayoutManagerInterface;
+use Magento\Cms\Model\PageFactory;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Cms Page Service Class
+ */
+class PageService
```

Comment:
- PageService can be unit tested for the logic of getPageById, createPageFactory, and fetchAvailableCustomLayouts. 
- Integration tests could ensure data provider works as expected in edit page scenarios.